AEPServices/Sources/utility/AnyCodable.swift

```diff
@@ -251,11 +251,38 @@ extension AnyCodable: Equatable {
 
 // MARK: Codable Helpers
 extension Encodable {
-    public func asDictionary(dateEncodingStrategy: JSONEncoder.DateEncodingStrategy = .deferredToDate
-    ) -> [String: Any]? {
+    /// Converts the value to a `[String: Any]` dictionary using JSON encoding.
+    ///
+    /// - Parameter dateEncodingStrategy: The strategy to use for encoding `Date` values.
+    ///   Defaults to `.deferredToDate`.
+    /// - Returns: A dictionary representation of the encoded value, or `nil` if encoding fails.
+    public func asDictionary(dateEncodingStrategy: JSONEncoder.DateEncodingStrategy = .deferredToDate) -> [String: Any]? {
         let encoder = JSONEncoder()
         encoder.dateEncodingStrategy = dateEncodingStrategy
         guard let data = try? encoder.encode(self) else { return nil }
         return (try? JSONSerialization.jsonObject(with: data, options: .allowFragments)).flatMap { $0 as? [String: Any] }
     }
 }
+
+extension Decodable {
+    /// Creates an instance of `Self` from a `[String: Any]` dictionary.
+    ///
+    /// - Parameters:
+    ///   - dictionary: The source dictionary.
+    ///   - dateDecodingStrategy: How `Date` values should be decoded.
+    ///     Defaults to `.deferredToDate`.
+    /// - Returns: A fully-decoded instance of `Self`, or `nil` if decoding fails.
+    static func fromDictionary(
```

Comment:
where all do we intend to use this?  only inside of core?  just asking because i noticed it's not public and wondered if that was intentional or not :)

Reply:
@timkimadobe  I think this was added to be used in Live Activity code (Messaging extension)
Do you modify the access to public ?